### ISSUE TITLE
Question on Principals

### ISSUE THREAD

**Issue by Stoffo** [sabre-io/dav#1121]
I already have my custom Backend-Classes set up and can read, delete and update contacts. Works fine so far. 

I read the docs, but I still have a question regarding the principal logic. MUST I inject the `Sabre\DAVACL\PrincipalCollection` in to the `$server` or is the `Sabre\CardDAV\AddressBookRoot` sufficient?

When I try to connect with a CardDAV-Client like contacts.app on OS/X for example, it just doesn't work:

![image](https://user-images.githubusercontent.com/7805224/48953506-5ce41200-ef46-11e8-9649-f030d400e938.png)

What's happening is a PROPFIND on /carddav/:

```xml
<A:propfind xmlns:A="DAV:"> 
	<A:prop> 
		<A:current-user-principal /> 
		<A:principal-URL /> 
		<A:resourcetype /> 
	</A:prop> 
</A:propfind>
```

Response:

```xml
<d:multistatus xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns">
	<d:response>
		<d:href>/carddav/</d:href>
		<d:propstat>
			<d:prop>
				<d:resourcetype>
					<d:collection />
				</d:resourcetype>
			</d:prop>
			<d:status>HTTP/1.1 200 OK</d:status>
		</d:propstat>
	</d:response>
</d:multistatus>
```

Am i missing something here? I just want a setup with one user who has one addressbook, no fancy ACL Logic whatsoever...

**Comment by evert** [comment]
CardDAV has a requirement that principals work, as well as the DAVACL plugin.

**Comment by Stoffo** [comment]
Yeah, I have implemented all methods like the docs say, but I still can't get it to work properly...

This is the output of the Browser Plugin:
![image](https://user-images.githubusercontent.com/7805224/49069368-682a8c80-f229-11e8-93c4-b8df4885e621.png)

The type is always "Unknown" and not "principal". What can I do to make this work?

And when I want to read the Principal I get this error:
`User did not have the required privileges ({DAV:}read) for path "principals/user"`
